test/ngAnimate/rafSchedulerSpec.js

```diff
@@ -0,0 +1,142 @@
+  }));
+
+  it('should run tasks based on how many RAFs have run in comparison to the task index',
+    inject(function($$rAFScheduler, $$rAF) {
+
+    var i, tasks = [];
+
+    for (i = 0; i < 5; i++) {
+      tasks.push([jasmine.createSpy()]);
+    }
+
+    $$rAFScheduler(tasks);
+
+    for (i = 1; i < 5; i++) {
+      var taskSpy = tasks[i][0];
+      expect(taskSpy).not.toHaveBeenCalled();
+      $$rAF.flush();
+      expect(taskSpy).toHaveBeenCalled();
+    }
+  }));
+
+  it('should space out subarrays by a RAF and run the internals in parallel',
+    inject(function($$rAFScheduler, $$rAF) {
+
+    var spies = {
+      a: spy('a'),
+      b: spy('b'),
+      c: spy('c'),
+
+      x: spy('x'),
+      y: spy('y'),
+      z: spy('z')
+    };
+
+    var items = [[spies.a, spies.x],
+                 [spies.b, spies.y],
+                 [spies.c, spies.z]];
+
+    $$rAFScheduler(items);
+    expect(spies.a).toHaveBeenCalled();
+    expect(spies.x).toHaveBeenCalled();
+
+    $$rAF.flush();
+
+    expect(spies.b).toHaveBeenCalled();
+    expect(spies.y).toHaveBeenCalled();
+
+    $$rAF.flush();
+
+    expect(spies.c).toHaveBeenCalled();
+    expect(spies.z).toHaveBeenCalled();
+
+    function spy(c) {
+      return jasmine.createSpy();
+    }
+  }));
+
+  describe('.waitUntilQuiet', function() {
+
+    it('should run the `last` provided function when a RAF fully passes',
+      inject(function($$rAFScheduler, $$rAF) {
+
+      var q1 = jasmine.createSpy();
+      $$rAFScheduler.waitUntilQuiet(q1);
+
+      expect(q1).not.toHaveBeenCalled();
+
+      var q2 = jasmine.createSpy();
+      $$rAFScheduler.waitUntilQuiet(q2);
+
+      expect(q1).not.toHaveBeenCalled();
+      expect(q2).not.toHaveBeenCalled();
+
+      var q3 = jasmine.createSpy();
+      $$rAFScheduler.waitUntilQuiet(q3);
+
+      expect(q1).not.toHaveBeenCalled();
+      expect(q2).not.toHaveBeenCalled();
+      expect(q3).not.toHaveBeenCalled();
+
+      $$rAF.flush();
+
+      expect(q1).not.toHaveBeenCalled();
+      expect(q2).not.toHaveBeenCalled();
+      expect(q3).toHaveBeenCalled();
+    }));
+
+    it('should always execute itself before the next RAF task tick occurs',
+      inject(function($$rAFScheduler, $$rAF) {
+
+      var log = [];
+
+      var quietFn = logFactory('quiet');
+      var tasks = [
+        [logFactory('task1')],
+        [logFactory('task2')],
+        [logFactory('task3')],
+        [logFactory('task4')]
+      ];
+
+      $$rAFScheduler(tasks);
+      expect(log).toEqual(['task1']);
+
+      $$rAFScheduler.waitUntilQuiet(quietFn);
+      expect(log).toEqual(['task1']);
+
+      $$rAF.flush();
+
+      expect(log).toEqual(['task1', 'quiet', 'task2']);
+
+      $$rAF.flush();
+
+      expect(log).toEqual(['task1', 'quiet', 'task2', 'task3']);
+
+      $$rAFScheduler.waitUntilQuiet(quietFn);
+
+      $$rAF.flush();
+
+      expect(log).toEqual(['task1', 'quiet', 'task2', 'task3', 'quiet', 'task4']);
+
+      function logFactory(token) {
```

Comment:
you can add `module(provideLog)` and then `inject(log)` to get logging

see: [provideLog](https://github.com/angular/angular.js/blob/2f6b6fb7a1dee0ff97c5d2959b927347eeda6e8b/test/helpers/testabilityPatch.js#L289-L326), [usage](https://github.com/angular/angular.js/blob/291d7c467fba51a9cb89cbeee62202d51fe64b09/test/ngRoute/routeSpec.js#L82)